### PR TITLE
pistol: add module

### PR DESCRIPTION
**Pull request by mtoohey31**
### Description

Adds a module for the [pistol file previewer](https://github.com/doronbehar/pistol). Pistol's configuration syntax is very simple and translates well into an attrset from mime type or path to command string.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -457,3 +457,6 @@ Makefile                                              @thiagokokada
 /modules/programs/ion.nix                             @jo1gi
 
 /modules/services/plex-mpv-shim.nix                   @starcraft66
+
+/modules/programs/pistol.nix                          @mtoohey31
+/tests/modules/programs/pistol                        @mtoohey31
```
Comment (sumnerevans) [inline .github/CODEOWNERS:462]: These are supposed to be alphabetized (looks like a few slipped past us in the past). Please sort theses in the proper place in the file.

File: modules/programs/pistol.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+let
+
+  cfg = config.programs.pistol;
+
+in {
+  meta.maintainers = [ maintainers.mtoohey ];
+
+  options.programs.pistol = {
+    enable = mkEnableOption
+      "General purpose file previewer designed for Ranger, Lf to make scope.sh redundant";
```
Comment (sumnerevans) [inline modules/programs/pistol.nix:14]:
```suggestion
      "pistol, a general purpose file previewer designed for terminal file managers";
```

File: modules/programs/pistol.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    config = mkOption {
+      type = with types; attrsOf str;
+      default = { };
```
Comment (sumnerevans) [inline modules/programs/pistol.nix:18]:
```suggestion
      default = { };
      defaultText = literalExpression '{ }';
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (mtoohey31) [reply .github/CODEOWNERS]: Ah, mb I missed that, fixed in 8f4ebfb21d6012647d9cb78e15e071ffc14d96f5.

**Review by sumnerevans — APPROVED** [review]

File: modules/programs/pistol.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      defaultText = literalExpression "{ }";
+      example = {
+        "text/*" =
+          "bat --paging=never --color=always --style=auto --wrap=character --terminal-width=%pistol-extra0% --line-range=1:%pistol-extra1% %pistol-filename%";
+        "application/json" =
+          "bat --paging=never --color=always --style=auto --wrap=character --terminal-width=%pistol-extra0% --line-range=1:%pistol-extra1% %pistol-filename%";
+      };
+      description = ''
+        Pistol configuration.
```
Comment (sumnerevans) [inline modules/programs/pistol.nix:27]: nit:

```suggestion
        Pistol configuration written to <filename>$XDG_CONFIG_HOME/pistol/pistol.conf</filename>.
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
Actually, this also needs an entry in the news file.

**Comment by mtoohey31** [comment]
> Actually, this also needs an entry in the news file.

Added in f2b4f203c0cc9841b5adf05d473a14eb8f650086.

**Comment by rycee** [comment]
Thanks! I merged it to master with a few stylistic changes.

**Pull request merged**